# 재귀호출함수

In [35]:
# Factorial을 계산하는 일반 함수
def factorial():
    number = int(input('숫자를 입력하세요.: '))
    result = 1
    for i in range(1, number + 1):
        result *= i
    return result

In [36]:
factorial()

120

In [31]:
# Factorial을 계산하는 Recursion 함수
def factorial_recursion(number):
    if number == 1:
        return 1
    return number * factorial_recursion(number - 1)

In [32]:
number = int(input('숫자를 입력하세요.: '))
factorial_recursion(number)

120

number에 5가 넘어왔을 때 함수가 실행되는 순서
5 * factorial_recursion(5)
5 * factorial_recursion(4)
5* (4 * factorial_recursion(3))
5* (4 * (3 * factorial_recursion(2)))
5* (4 * (3 * (2 * factorial_recursion(1))))
5* (4 * (3 * (2 * (1))))
***
5* (4 * (3 * (2 * 1)))
5* (4 * (3 * 2))
5* (4 * 6)
5 * 24
120